src/Components/Result.jsx

```diff
@@ -1,23 +1,30 @@
-import React from "react";
+import { useRef } from "react";
 
-const Result = ({ userNum, computerNum }) => {
-  let result;
-  if (userNum) {
-    if (computerNum > userNum) {
-      result = "Your guess is too low!";
-    } else if (computerNum < userNum) {
-      result = "Your guess is too high";
-    } else if (computerNum == userNum) {
-      result = "Yay! You guessed correctly!";
-    } else {
-      result = "Enter valid input";
-    }
+const Result = ({ userNum }) => {
+  const randomNum = useRef(Math.floor(Math.random() * 100) + 1);
+  console.log(randomNum);
+
+  if (userNum < 0) {
+    return <h3>Enter valid input</h3>;
   }
+
+  let result = "";
+
+  if (randomNum.current > userNum) {
+    result = "Your guess is too low!";
+  } else if (randomNum.current < userNum) {
+    result = "Your guess is too high";
+  } else if (randomNum.current === userNum) {
+    result = "Yay! You guessed correctly!";
+  } else {
+    result = "Enter valid input!";
```

Comment:
The code will never arrive here because if the input is NaN, userNum will be -1, meaning invalid input will ALWAYS end up on line 8 above. So we can remove this else case. In fact, you don't need any else cases. Can just be a set of if cases like:

```
if (randomNum.current > userNum) {
    result = "Your guess is too low!";
  } 
  
if (randomNum.current < userNum) {
    result = "Your guess is too high";
  }

 if (randomNum.current === userNum) {
    result = "Yay! You guessed correctly!";
  }
```